src/main/java/spoon/template/Substitution.java

```diff
@@ -684,15 +684,16 @@ public static void redirectTypeReferences(CtElement element, CtTypeReference<?>
 //				}
 
 				// contract: if a proxy parameter is declared and named "x" (@Parameter("x")), then a type member named "x" must exist.
-				boolean found = false;
-				for (CtTypeMember member: c.getTypeMembers()) {
-					if (member.getSimpleName().equals(proxyName)) {
-						found = true;
-					}
-				}
-				if (!found) {
-					throw new TemplateException("if a proxy parameter is declared and named \"" + proxyName + "\", then a type member named \"\" + proxyName + \"\" must exist.");
-				}
+				// it is not good contract, because fields of inner classes and method invocations cannot be handled then
```

Comment:
I know what you mean, but the reason why I need to relax this constrain is not the name conflict but the ability to tell Spoon that I do not want to replace string literal but I want to replace simple name of the field or method invocation.
Try to rollback this [change](https://github.com/INRIA/spoon/pull/1476/commits/8ee34001108a310ed04ee0070417fee95e752df2) using code of this PR and then run that test. It will fail and if you try to fix it, then you understand what I mean.
The code
```java
void invoke() {
 	iface.$method$();
 }
```
is not replace to 
```java
void invoke() {
 	iface.toBeInvokedMethod();
 }
```
but is replaced with 
```java
void invoke() {
 	"toBeInvokedMethod";
 }
```
and the last one is correct from C2, C3 point of view, because the parameter value is CtLiteral. So I need to force the system to keep parameter value of type String! But how if the constraint is so strict?